File: src/main/java/de/schramm/royalbash/controller/service/core/card/NoOpCard.java
package de.schramm.royalbash.controller.service.core.card;

import de.schramm.royalbash.controller.service.core.Card;
import de.schramm.royalbash.controller.service.core.Context;
import de.schramm.royalbash.controller.service.core.Game;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class NoOpCard implements Card {

    private final String id;
    private final String name;
    private final int cost;

    @Override
    public Game invoke(Context context) {
        return context.getGame();
    }

    @Override
    public boolean isPlaceableOnSpot() {
        return false;
    }
}

File: src/main/java/de/schramm/royalbash/controller/service/core/card/NoOpCard.kt
package de.schramm.royalbash.controller.service.core.card

import de.schramm.royalbash.controller.service.core.Card
import de.schramm.royalbash.controller.service.core.Context
import de.schramm.royalbash.controller.service.core.Game

class NoOpCard (
        override val id: String,
        override val name: String = "NoOp Card",
        override val text: String = "",
        override val cost: Int = 0
) : Card {

    override fun invoke(context: Context): Game {
        return context.game
    }
}
